src/main/java/com/excelsiorjet/api/tasks/JetProject.java

```diff
@@ -643,13 +641,116 @@ public JetHome validate(boolean validateForBuild) throws JetTaskFailureException
             execProfilesName = projectName;
         }
         if (validateForBuild) {
-            validateForBuild(jetHomeObj);
+            validateForBuild(excelsiorJet);
         }
 
-        return jetHomeObj;
+        processDependencies();
     }
 
-    private void validateForBuild(JetHome jetHomeObj) throws JetTaskFailureException {
+    public void processDependencies() throws JetTaskFailureException {
+        for (DependencySettings dependency : dependencies) {
+            if (!dependency.isValid()) {
+                throw new JetTaskFailureException(s("JetApi.InvalidDependencySetting", dependency.idStr()));
+            }
+        }
+        List<DependencySettings> dependenciesSettings = dependencies.stream()
+                .filter(DependencySettings::isSettings)
+                .collect(toList());
+        List<DependencySettings> externalDependencies = dependencies.stream()
+                .filter(DependencySettings::isExternal)
+                .collect(toList());
+        if (externalDependencies == null) {
+            externalDependencies = Collections.emptyList();
+        }
+        if (dependenciesSettings == null) {
+            dependenciesSettings = Collections.emptyList();
+        }
+
+        for (DependencySettings externalDependency : externalDependencies) {
+            if (externalDependency.path.isDirectory() && (externalDependency.pack != null && !PackingType.NONE.userValue.equals(externalDependency.pack))) {
+                throw new JetTaskFailureException(s("JetApi.NotPackedExternalDirectory", externalDependency.path));
+            }
+        }
+
+        for (DependencySettings dependencySettings : dependenciesSettings) {
+            List<ProjectDependency> matchedDependencies = dependenciesMatchedBy(dependencySettings);
```

Comment:
If ignoreProjectDependencies is set, we should not match depenedencySettings upon projectDependencies except main jar. I think we should just check that there is no non-extrernal dependencies except main Jar there with additional pass over dependency settings.